### ISSUE TITLE
Period Results template

### ISSUE THREAD

**Issue by paddyow** [Kademi/kademi-public#156]
This user https://adminupdates.admin.kademi-qa.co/manageUsers/34328582/#summary-tab
This template http://test-adminupdates.kademi-qa.co/period-results?periodId=34417286

Should the topic not be automatically selected based on the period the user chooses from the previous page when they click on 'view results' button?

Previous screen:
![image](https://user-images.githubusercontent.com/2201340/114338583-b4768d00-9b96-11eb-86f5-a547aba7f1aa.png)

Keditor:
![image](https://user-images.githubusercontent.com/2201340/114338259-0f5bb480-9b96-11eb-8270-437989c36699.png)

Leaderboard is showing the current period, but should show a past period as above where the topic and period should be dynamic based on which period and topic the user selected from the previous screen.

Past results is showing data for a different topic so this is incorrect. 

UI artwork
![image](https://user-images.githubusercontent.com/2201340/114338206-ed623200-9b95-11eb-8466-7f39473a8b15.png)

**Comment by bradmac** [comment]
> Should the topic not be automatically selected based on the period the user chooses from the previous page when they click on 'view results' button?

Yeah, the topic should be determined by the period they're viewing

So @linh please remove the topic setting and ensure the component just uses the period to get the topic

> Leaderboard is showing the current period, but should show a past period as above where the topic and period should be dynamic based on which period and topic the user selected from the previous screen.

For the kpiLeaderboardWebRecog component settings, the topic select box should have an option for "Automatic", which means that the topic will be determined from the period (same as above). 

And if there is a period in context then it should show the leaderboard for that period.

Note that this component can be used standalone (eg on the dashboard) so it needs to continue to allow a user to select a topic, nd if not period is in context it should show the current period.

> Past results is showing data for a different topic so this is incorrect. Should be..

Yep, needs to use the topic from the period in context.
